nc/source/graphics/resources/GraphicsResources.cpp

```diff
@@ -122,14 +127,79 @@ namespace nc::graphics
         return m_accessors.at(uid);
     }
 
+    void TexturesData::AddTexture(Graphics* graphics, Texture texture)
```

Comment:
I think these Add methods need to enforce the max size isn't exceeded. Without it there's UB in UpdateTextures.